### PR TITLE
Hydrate required associations when hydrate object

### PR DESCRIPTION
**Pull request by vincentchalamon**
When I want to create an entity but I don't care about it's data, I need to create manually all its required associations. This can be done through the hydrator.

For example, to create a User with a required association Company:

``` gherkin
Feature: I need to create an entity
    Scenario: I can create an entity
        Given the following company:
            | name      |
            | KnpLabs |
        And the following user:
            | company |
            | KnpLabs |
        ...
```

I need to create the company manually, name it, and link it manually to the user. This PR allows to create dynamically **required single associations** (ManyToMany & OneToMany are not concerned) automatically:

``` gherkin
Feature: I need to create an entity
    Scenario: I can create an entity
        Given there is 1 user
        ...
```

### CODE REVIEW

**Comment by pilot** [comment]
:+1:

**Comment by guillaumevoisin** [comment]
:+1:

**Comment by Gregcop1** [comment]
:+1:

**Comment by vincentchalamon** [comment]
Ping @PedroTroller @iamluc, can somebody review this PR please ? :smile:

File: src/Knp/FriendlyContexts/Doctrine/EntityHydrator.php [hunk 1]
```diff
@@ -82,6 +82,33 @@ public function completeFields(ObjectManager $em, $entity)
...
+                !isset($mapping['joinColumns'][0]['nullable']) ||
+                $mapping['joinColumns'][0]['nullable'] === true ||
+                $accessor->getValue($entity, $associationName) !== null
+            ) {
+                continue;
+            }
+            try {
+                // Create association object
+                $relatedClass = $metadata->getAssociationTargetClass($associationName);
+                $property = new $relatedClass;
```
Comment (Einenlum) [inline src/Knp/FriendlyContexts/Doctrine/EntityHydrator.php]: What if the object requires constructor arguments?

Maybe we should use [ReflectionClass::newInstanceWithoutConstructor](http://php.net/manual/en/reflectionclass.newinstancewithoutconstructor.php).

**Comment by Einenlum** [comment]
:+1: 

@Nek- What do you think?

File: src/Knp/FriendlyContexts/Doctrine/EntityHydrator.php (hunk 1 above)
Comment (Nek-) [inline src/Knp/FriendlyContexts/Doctrine/EntityHydrator.php]: That's a very good warning... But it's already done like that : https://github.com/vincentchalamon/FriendlyContexts/blob/master/src/Knp/FriendlyContexts/Context/EntityContext.php#L24

I think it's ok to have it here again. My only regret is that it looks like a piece of copy/paste.

**Comment by vincentchalamon** [comment]
Ping @Nek- @Einenlum

**Pull request closed without merging**